taxonomy/openai/client.py

```diff
@@ -1,25 +1,31 @@
-"""openai client"""
+"""CHAT_COMPLETION_API client"""
+import json
+import logging
 
-import openai
+import requests
 
 from django.conf import settings
 
-openai.api_key = settings.OPENAI_API_KEY
+log = logging.getLogger(__name__)
 
 
 def chat_completion(prompt):
     """
-    Use chatGPT https://api.openai.com/v1/chat/completions endpoint to generate a response.
-
+    Pass message list to chat endpoint, as defined by the CHAT_COMPLETION_API setting.
     Arguments:
         prompt (str): chatGPT prompt
     """
-    response = openai.ChatCompletion.create(
-        model="gpt-3.5-turbo",
-        messages=[
-            {"role": "user", "content": prompt},
-        ]
+    completion_endpoint = settings.CHAT_COMPLETION_API
+    completion_endpoint_key = settings.CHAT_COMPLETION_API_KEY
+    headers = {'Content-Type': 'application/json', 'x-api-key': completion_endpoint_key}
+    connect_timeout = getattr(settings, 'CHAT_COMPLETION_API_CONNECT_TIMEOUT', 1)
+    read_timeout = getattr(settings, 'CHAT_COMPLETION_API_READ_TIMEOUT', 15)
+    body = {'message_list': [{'role': 'assistant', 'content': prompt},]}
+    response = requests.post(
+        completion_endpoint,
+        headers=headers,
+        data=json.dumps(body),
+        timeout=(connect_timeout, read_timeout)
```

Comment:
nit: It would be more readable to use `settings.CHAT_COMPLETION_API_CONNECT_TIMEOUT` and `settings. CHAT_COMPLETION_API_READ_TIMEOUT ` directly here.